blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/outputs/handlers/HttpRollupHandlerWithESIntegrationTest.java

```diff
@@ -90,7 +90,7 @@ public void setup() throws Exception {
         writer.insertFull(metrics);
 
         httpHandler = new HttpRollupsQueryHandler();
-        ((ElasticIO)QueryDiscoveryModuleLoader.getDiscoveryInstance()).setClient(esSetup.client());
+        ((ElasticIO) GenericClassLoader.getGenericInstance(DiscoveryIO.class, CoreConfig.DISCOVERY_MODULES)).setClient(esSetup.client());
 
```

Comment:
Let's try adding some caching logic here, ie extract ((ElasticIO) GenericClassLoader.getGenericInstance(DiscoveryIO.class, CoreConfig.DISCOVERY_MODULES)) as a private static field. Is it doable?